pkg/utils/formatting.go

```diff
@@ -51,16 +56,23 @@ func RenderDisplayStrings(displayStringsArr [][]string, columnAlignments []Align
 			Width:     padWidth,
 			Alignment: alignment,
 		}
+		columnPositions[i+1] = columnPositions[i] + padWidth + 1
 	}
-	output := getPaddedDisplayStrings(displayStringsArr, columnConfigs)
-
-	return output
+	// Add the removed columns back into columnPositions (a removed column gets
+	// the same position as the following column); clients should be able to rely
+	// on them all to be there
+	for _, removedColumn := range removedColumns {
```

Comment:
I have a heuristic that says if we do something and then revert what we did, there exists a simpler solution where we don't need to revert anything. I've spent some time seeing if I can simplify things by adding a 'Hide' field to ColumnConfig and using column configs for everything rather than having a separate padWidths variable, but I haven't managed to get tests passing and the code seems to be more complicated, so I'm happy to leave it

Reply:
I agree, and I didn't like having to put things back that we removed earlier. However, I spent a lot of time on this, and tried several other approaches (e.g. reordering code so that we calculate the pad widths first), but none of them really worked out, so I settled on this in the end, not feeling great about it.